test/core/xds/xds_transport_fake.h

```diff
@@ -128,11 +131,15 @@ class FakeXdsTransportFactory : public XdsTransportFactory {
     bool status_sent_ ABSL_GUARDED_BY(&mu_) = false;
     bool orphaned_ ABSL_GUARDED_BY(&mu_) = false;
     size_t reads_started_ ABSL_GUARDED_BY(&mu_) = 0;
-    bool read_pending_ ABSL_GUARDED_BY(&mu_) = false;
+    size_t num_pending_reads_ ABSL_GUARDED_BY(&mu_) = 0;
     std::deque<std::string> to_client_messages_ ABSL_GUARDED_BY(&mu_);
+    std::function<void()> too_many_pending_reads_callback_;
```

Comment:
Similar to my comment below, I don't think we need to store a copy of the callback in `FakeStreamingCall`.  Instead, we can access it via `transport_->factory_->too_many_pending_reads_callback_`.

Reply:
Done. Except I added an accessor for factory as the field is private.